packages/core/src/model.types.ts

```diff
@@ -28,12 +29,18 @@ export interface Model<
   TModelCreators = void
 > {
   initialContext: TContext;
-  assign: <TEventType extends TEvent['type'] = TEvent['type']>(
+  assign<TEventType extends TEvent['type'] = TEvent['type']>(
     assigner:
       | Assigner<TContext, ExtractEvent<TEvent, TEventType>>
       | PropertyAssigner<TContext, ExtractEvent<TEvent, TEventType>>,
     eventType?: TEventType
-  ) => AssignAction<TContext, ExtractEvent<TEvent, TEventType>>;
+  ): AssignAction<TContext, ExtractEvent<TEvent, TEventType>>;
+  assign(
+    assigner:
+      | Assigner<TContext, DoneInvokeEvent<any>>
+      | PropertyAssigner<TContext, DoneInvokeEvent<any>>,
+    eventType: `done.invoke` | `done.invoke.${string}`
```

Comment:
The syntax for this has only been introduced in TS 4.1 (not even a year ago). Introducing this here would break all projects using older TS versions (which is very unfortunate :/)

Reply:
Since that string doesn't really matter at all - I think you could go with this:
```suggestion
    eventType: 'done.invoke'
```

Reply:
What happens if this is used in an older version of TS? Does it just not compile?

Reply:
I haven't rechecked it but I think that TS would not be able to create any types based on this file - it's effectively a syntax error for it. You can check out how `.D.TS` tab is empty on the right of this playground (it's setup to pre-4.1 version):
https://www.typescriptlang.org/play?ts=4.0.5#code/C4TwDgpgBAglC8UAGwIGdhIFCklAQgsqhgCQDeMAvtrtAMJEDkJwATE0A

Which in turn would make models unusable (or typed to any or smth like that) on those TS versions

Reply:
7cdddb48a75836c0d1f5059f6a430ab0300c8d82